src/axom/primal/operators/detail/intersect_bounding_box_impl.hpp

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef PRIMAL_INTERSECT_BOUNDING_BOX_HPP_
+#define PRIMAL_INTERSECT_BOUNDING_BOX_HPP_
+
+#include "axom/core/Macros.hpp"
+
+#include <type_traits> // for std::is_floating_point< T >()
+
+namespace axom
+{
+namespace primal
+{
+namespace detail
+{
+
+/*!
+ * \brief Helper routine for AABB / AABB intersection test.
+ *
+ * \param [in] min1 the 1st AABB min coordinate along a direction.
+ * \param [in] max1 the 1st AABB max coordinate along a drection.
+ * \param [in] min2 the 2nd AABB min coordinate along a direction.
+ * \param [in] max2 the 2nd AABB max coordinate along a drection.
+ *
+ * \return status true if the AABB's intersect, otherwise false.
+ *
+ * \note This routine is called by the intersect AABB/AABB methods for each
+ *  spatial dimension.
+ */
+template < typename T >
+AXOM_HOST_DEVICE
+inline bool intersect_bbox_bbox_test( const T& min1,
+                                      const T& max1,
+                                      const T& min2,
+                                      const T& max2 )
+{
+  AXOM_STATIC_ASSERT( std::is_floating_point< T >::value );
+  
+  bool status = ( ( max1 < min2 || min1 > max2 ) ? false : true );
+  return status;
+}
+
+/*!
+ * \brief Checks if a specified bounding box intersects with the supplied
+ *  bounding box.
+ *
+ * \param [in] xmin1 x-coordinate of the specified lower bounding box corner
+ * \param [in] xmax1 x-coordinate of the specified upper bounding box corner
+ * \param [in] ymin1 y-coordinate of the specified lower bounding box corner
+ * \param [in] ymax1 y-coordinate of the specified upper bounding box corner
+ * \param [in] xmin2 x-coordinate of the supplied lower bounding box corner
+ * \param [in] xmax2 x-coordinate of the supplied upper bounding box corner
+ * \param [in] ymin2 y-coordinate of the supplied lower bounding box corner
+ * \param [in] ymax2 y-coordinate of the supplied upper bounding box corner
+ *
+ * \return status true if the bounding boxes intersect, otherwise, false.
+ */
+template < typename T >
+AXOM_HOST_DEVICE
+inline bool intersect_bounding_box( const T& xmin1,
+                              			const T& xmax1,
+                           	    		const T& ymin1,
+                               			const T& ymax1,
+                           			    const T& xmin2,
+                           			    const T& xmax2,
+                           			    const T& ymin2,
+                           			    const T& ymax2 )
```

Comment:
nitpick: code formatting of the arguments looks odd. Do you have any tabs in this file?

You may want to run this through uncrustify but don't let uncrustify do anything weird with the lambdas

Reply:
Done!